### ISSUE TITLE
SUMO-based language localization

### ISSUE THREAD

**Issue by jefft0** [ontologyportal/Calendar#3]
Hi Adam,

The calendar app displays strings like "Monday". To localize these strings for other languages, a first instinct would be to use Java's localization support. But SUMO already has string localization, like the French word for Monday:
https://github.com/ontologyportal/sumo/blob/master/Translations/french_format.kif#L969

For many other terms, we would be relying on SUMO's language settings. What do you think about doing this for the Calendar display elements too, like weekday and month names?

**Comment by apease** [comment]
Hi Jeff,
   I agree - it makes for a much better demo to use SUMO's facilities. 
Even better than the termFormat content might be to use the links to 
Open Multilingual Wordnet since it has more languages, and a much larger 
vocabulary.  We could start with days of the week, months, holidays but 
later expand to many other things like appointment type names and 
location names. Check out com.articulate.sigma.OMWordnet

Adam

On 06/13/2016 12:41 PM, Jeff Thompson wrote:

> Hi Adam,
> 
> The calendar app displays strings like "Monday". To localize these
> strings for other languages, a first instinct would be to use Java's
> localization support. But SUMO already has string localization, like the
> French word for Monday:
> https://github.com/ontologyportal/sumo/blob/master/Translations/french_format.kif#L969
> 
> For many other terms, we would be relying on SUMO's language settings.
> What do you think about doing this for the Calendar display elements
> too, like weekday and month names?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/ontologyportal/Calendar/issues/3, or mute the
> thread
> https://github.com/notifications/unsubscribe/ACBvHUjmVcZIcl3SEfza-UdCDiZBv362ks5qLbJcgaJpZM4I0qmT.

## 

---

Adam Pease
http://www.ontologyportal.org
http://www.articulatesoftware.com
http://www.adampease.org

**Comment by jefft0** [comment]
I'll take a look at the WordNet links. Different localities would say today is "13 June" vs. "June 13". Does WordNet have a way express this kind of detail? (Not a big deal if it doesn't.)

**Comment by arademaker** [comment]
No, wordnet does not have date patterns explicit. But you can look at Stanford http://nlp.stanford.edu/software/sutime.html tool!

**Comment by apease** [comment]
I agree, for time/date formats SUtime is an excellent choice

On 06/13/2016 01:28 PM, Alexandre Rademaker wrote:

> No, wordnet does not have date patterns explicit. But you can look at
> Stanford http://nlp.stanford.edu/software/sutime.html tool!
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/ontologyportal/Calendar/issues/3#issuecomment-225698189,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACBvHZ_0OjSz30Xe-wfm47qSvzr57cKAks5qLb1TgaJpZM4I0qmT.

## 

---

Adam Pease
http://www.ontologyportal.org
http://www.articulatesoftware.com
http://www.adampease.org

**Comment by jefft0** [comment]
A quick glance at the SUTime page says "The currently available rules support only English." Is that right?

**Comment by apease** [comment]
Hi Jeff and Alexandre,
   Maybe I was too quick to endorse SUtime.  I see that it's only 
English, although there's still utility, and need, for handle the many 
different options that exist in English.  Check out 
com.articulate.sigma.semRewrite.datesandnumber and I think it's already 
integrated with SUtime, so it's a good place to start.
   After that though it's possible we might need to look at 
http://dbs.ifi.uni-heidelberg.de/index.php?id=129#c784 or others

Adam

On 06/13/2016 01:42 PM, Jeff Thompson wrote:

> A quick glance at the SUTime page says "The currently available rules
> support only English." Is that right?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/ontologyportal/Calendar/issues/3#issuecomment-225702147,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACBvHTNNxGcYit3pNkXg_6g6QpA_9Plzks5qLcC_gaJpZM4I0qmT.

## 

---

Adam Pease
http://www.ontologyportal.org
http://www.articulatesoftware.com
http://www.adampease.org
